buildscripts/executeSeleniumTestsReleaseBuild.sh

```diff
@@ -0,0 +1,198 @@
+echo Updating testng.xml
+sed -i.bak 's/4000/4001/g' testng.xml    # Documentation wil be published on port 4001
+sed -i.bak 's/4444/4445/g' testng.xml    # The Selenium Grid hub process will use port 4445
+
+# Create the latest ux-aspects-build image if it does not exist
+docker_image_build; echo
+echo Executing the Selenium tests in the $UX_ASPECTS_BUILD_IMAGE_NAME:$UX_ASPECTS_BUILD_IMAGE_TAG_LATEST container
+
+# Remove any ContainerID file left behind by the previous container
+if [ -f "$PWD/ContainerIDReleaseBuild" ]; then
+    ContainerID=$(<"$PWD/ContainerIDReleaseBuild")
+    echo Previous container ID is $ContainerID
+    if [ ! -z $ContainerID ]; then
+        echo Killing container ID $ContainerID
+    	docker kill $ContainerID
+        echo Removing container ID $ContainerID
+    	docker rm -f $ContainerID
+        echo Removing container ID file
+    	rm -f "$PWD/ContainerIDReleaseBuild"
+    fi
+fi
+
+# Remove the file indicating that the tests have finished
+if [ -f "$PWD/GridHubFinishedReleaseBuild" ]; then
+    echo Removing grid hub flag
+    rm -f "$PWD/GridHubFinishedReleaseBuild"
+fi
+
+# Create a new container which will build the new web service
+echo Starting new container
+docker_image_run bash buildscripts/executeSeleniumTestsReleaseBuildDocker.sh &
+
+# Loop until the container has been created
+containerIDCheckDelay=5
+while :
+do
+	if [ -f "$PWD/ContainerIDReleaseBuild" ]; then
+		ContainerID=$(<"$PWD/ContainerIDReleaseBuild")
+		echo New container ID is $ContainerID
+		break
+	else
+		echo No container ID file created yet
+	fi
+	echo Sleeping for $containerIDCheckDelay seconds
+    sleep $containerIDCheckDelay
+done
+
+# Display the output of the container
+docker logs -f $ContainerID &
+
+# Loop until the web service is available
+wgetCheckDelay=5
+while :
+do
+    bash -c 'wget --proxy=off -t 1 -O webservice.html localhost:4001 ; exit $?'
+	wgstatus=$?
+    echo wget returned $wgstatus
+	rm -f "$PWD/webservice.html"
+    if [ $wgstatus -eq 0 ]; then
+        echo Web service is available
+        break
+    fi
+	echo Sleeping for $wgetCheckDelay seconds
+    sleep $wgetCheckDelay
+done
+
+# Kill any process using port 4445 (the Selenium Grid hub process) and start a new process
+echo Kill any existing Selenium Grid hub process
+PID_SELENIUM=`/usr/sbin/fuser -n tcp 4445 2> /dev/null`
+echo Old Selenium Grid hub process ID is $PID_SELENIUM
+if [ ! -z "$PID_SELENIUM" ] ; then echo "Killing existing Selenium Grid hub process" ; kill -9 $PID_SELENIUM ; fi
+sleep 10
+PID_SELENIUM=`/usr/sbin/fuser -n tcp 4445 2> /dev/null`
+echo After kill, Selenium Grid hub process ID is $PID_SELENIUM
+
+echo Start the new Selenium Grid hub process
+cd $HOME_FOLDER/ux-aspects
+rm -rf target
+cd $HOME_FOLDER/ux-aspects/configuration
+java -jar /home/UXAspectsTestUser/ux-aspects/Selenium/selenium-server-standalone-3.3.1.jar -role hub -hubConfig hub/hubConfigReleaseBuild.json &
+sleep 15
```

Comment:
Potentially a loop could be added here, making this part more resilient and possibly reducing the duration of the script. Can a loop function be written so that similar checks can reuse some code?